stan/math/prim/mat.hpp

```diff
@@ -46,6 +46,8 @@
 #include <stan/math/prim/mat/err/constraint_tolerance.hpp>
 #include <stan/math/prim/mat/err/validate_non_negative_index.hpp>
 
+#include <stan/math/prim/mat/vectorize/apply_scalar_unary.hpp>
```

Comment:
It needs to be before the `stan/math/prim/mat/fun/` files to avoid the exp error that we got earlier. After all, the tests for the vectorized functions include `stan/math/prim/mat/fun/mat.hpp` first because of the order of headers.